spec/system/reports_spec.rb

```diff
@@ -29,41 +29,43 @@
       fill_in 'タイトル', with: '初めての日報'
       fill_in '内容', with: <<~TEXT
         参考にした日報のURLを書きました。
-        http://localhost:3000/reports/#{@alice_report.id}
+        http://localhost:3000/reports/#{@report.id}
       TEXT
 
-      expect do
-        click_on '登録する'
-        assert_text '日報が作成されました。'
-      end.to change { ReportMention.count }.by(1)
+      click_on '登録する'
+      expect(page).to have_content '日報が作成されました。'
 
-      visit report_path(@alice_report)
+      expect(page).to have_selector '.show-item', text: '初めての日報'
+
+      click_on "http://localhost:3000/reports/#{@report.id}"
 
       expect(page).to have_selector '.mentions-container', text: '初めての日報'
     end
 
     scenario 'updating a report' do
-      visit report_path(@alice_report)
+      visit report_path(@report)
       click_on 'この日報を編集'
 
       fill_in 'タイトル', with: '最後の日報'
       fill_in '内容', with: <<~TEXT
         自分の日報のURLを書きました。
-        http://localhost:3000/reports/#{@alice_report.id}
+        http://localhost:3000/reports/#{@report.id}
       TEXT
       click_on '更新する'
       expect(page).to have_content '日報が更新されました。'
 
+      expect(page).to have_selector '.show-item', text: '最後の日報'
+
       expect(page).to have_no_selector '.mentions-container', text: '最後の日報'
     end
 
     scenario 'destroying a report' do
-      visit report_path(@alice_report)
+      visit report_path(@report)
 
       click_on 'この日報を削除'
       expect(page).to have_content '日報が削除されました。'
 
-      expect(page).to have_no_selector "#report_#{@alice_report.id}", text: 'アリスの日報'
+      expect(page).to have_no_selector "#report_#{@report.id}", text: 'アリスの日報'
```

Comment:
```suggestion
      expect(page).to have_current_path reports_path
      expect(page).to have_no_selector "#report_#{@report.id}", text: 'アリスの日報'
```

現在のパスが日報一覧のページであることを確認しておくとよりテストとして堅牢になります。
（全然無関係なページを表示していたら、日報の情報が出ないのは当たり前なので）

Reply:
期待する要素の存在を確認する前に、その要素が存在する画面に遷移しているかを確認するのが良いということですね。こちら修正しました！！

Reply:
```ruby
expect(page).to have_no_selector "#report_#{@report.id}"
```

もしくは

```ruby
expect(page).to have_no_content 'アリスの日報'
```

のどちらかだけでテストとしては十分かも（ユーザー視点なら後者かな？）